apps/remote_control/lib/lexical/remote_control/search/indexer/extractors/struct_definition.ex

```diff
@@ -0,0 +1,41 @@
+defmodule Lexical.RemoteControl.Search.Indexer.Extractors.StructDefinition do
+  alias Lexical.Document.Position
+  alias Lexical.Document.Range
+  alias Lexical.RemoteControl.Analyzer
+  alias Lexical.RemoteControl.Search.Indexer.Entry
+  alias Lexical.RemoteControl.Search.Indexer.Source.Reducer
+
+  def extract({:defstruct, _, [_fields]} = definition, %Reducer{} = reducer) do
+    document = reducer.analysis.document
+    block = Reducer.current_block(reducer)
+    {:ok, current_module} = Analyzer.current_module(reducer.analysis, Reducer.position(reducer))
+    range = range(document, definition)
+
+    entry =
+      Entry.definition(
+        document.path,
+        block,
+        current_module,
+        :struct,
+        range,
+        Application.get_application(current_module)
+      )
+
+    {:ok, entry}
+  end
+
+  def extract(_, _) do
+    :ignored
+  end
+
+  defp range(document, definition) do
```

Comment:
We do this in enough places that, IMO, it should really be a utility function like `Lexical.Ast.range_in_document(node, document)`.

This function could also raise an exception with a useful message if `Sourceror.get_range` returns nil for the given node (which it can), which will help us see the actual syntax it's failing on instead of whatever downstream match failure occurs because it returns nil instead of a map.

Reply:
I added `Lexical.Ast.Range.fetch` and `Lexical.Ast.Range.fetch!` for this.